src/file_pipe_log/mod.rs

```diff
@@ -6,6 +6,8 @@ mod pipe;
 mod pipe_builder;
 mod reader;
 
+pub use format::FileNameExt;
+pub use log_file::{build_file_writer, LogFd};
```

Comment:
No need to export them now?

Reply:
No,  Because `FileNameExt` and `LogFd`  will be used in test code of `engine.rs`

Reply:
as refer to `build_file_writer`, we can safely remove it

Reply:
```
        let dump_it = Engine::dump(file_id.build_file_path(dir.path()).as_path()).unwrap();
```
This will use `LogFd` and `FileNameExt `

Reply:
You can keep `FileNameExt`, `LogFd` can be removed?

Reply:
I think not
```rust


        //dump file
        let file_id = FileId {
            queue: LogQueue::Rewrite,
            seq: 1,
        };
        let dump_it = Engine::dump(file_id.build_file_path(dir.path()).as_path()).unwrap();
        assert!(0 == dump_it.map(|i| i.unwrap()).count());

        //dump dir that does not exists
        assert!(Engine::dump(Path::new("/not_exists_dir")).is_err());
```

`FileId` is used

Reply:
Yes, you are so careful 👍 
By the way, why `cargo clippy` and `cargo build` can't detect unused import ?

Reply:
Because it's a public export, even though we don't use it internally, rust thinks other users might use it from outside this library.

Reply:
😃